### ISSUE TITLE
Support for the Ninja Multi-Config generator

### ISSUE THREAD

**Issue by MathiasMagnus** [microsoft/vscode-cmake-tools#1423]
### Brief Issue Summary

CMake Tools bails out when trying to use the [Ninja Multi-Config generator](https://cmake.org/cmake/help/latest/generator/Ninja%20Multi-Config.html).

### Expected:

1. Set `"preferredGenerator": { "name": "Ninja Multi-Config" }` for a Kit
2. Develop much like on would with a multi-configuration generator, such as MSBuild

### Apparent Behavior:

Receive an error when setting the kit: `Unable to set kit "{0}".`

### CMake Tools Log

```
[main] Configuring folder: Test-Boost.UT 
[main] Unable to determine what CMake generator to use. Please install or configure a preferred generator, or update settings.json, your Kit configuration or PATH variable. Error: No usable generator found.
[main] Unable to determine what CMake generator to use. Please install or configure a preferred generator, or update settings.json, your Kit configuration or PATH variable. Error: No usable generator found.
```

### Platform and Versions

- **Operating System**: Windows
- **CMake Version**: 3.18.1
- **VSCode Version**: 1.48.2
- **CMake Tools Extension Version**: 1.4.1
- **Compiler/Toolchain**: Visual C++ 2019

**Comment by Wastus** [comment]
Ran into this issue as well, you can workaround this by setting: `"cmake.generator": "Ninja Multi-Config",` in the settings.json. Though this will not help if you want to switch between kits with different generators.
It's a bit strange that this works without complaints, but setting it in `preferredGenerator` does not.

As the `cmake.generator` settings overrides the `preferredGenerator` setting, you do not have to remove the content from the kit definition until the issue is resolved.

#### Platform and Versions
 * Operating System: WIndows
 * CMake Version: 3.19.4
 * VSCode Version: 1.53.0
 * CMake Tools Extension Version: 1.5.3
 * Compiler/Toolchain: ARM GCC 10.2.1

**Comment by lygstate** [comment]
Any plan on this?

**Comment by bobbrow** [comment]
We are considering it for 1.10.0

**Comment by bobbrow** [comment]
The fix for this issue is available in 1.10 "pre-release".  You can help us validate it by:
1. Opening the extensions panel in VS Code
2. Selecting the CMake Tools extension
3. Clicking the "Switch to Pre-release Version" button

![image](https://user-images.githubusercontent.com/12818240/156218538-7bce4a1f-4fc9-43cc-afb0-7f423aa5980e.png)

**Comment by bobbrow** [comment]
This was included in 1.10 which is now generally available.
